apps/dashboard/app/lib/cluster_cache.rb

```diff
@@ -0,0 +1,43 @@
+#frozen_string_literal: true
+
+module ClusterCache
+  def cluster_options
+    Rails.cache.fetch('script_cluster_options', expires_in: 4.hours) do
+      cluster_max_cores.map do |cluster_id, max|
+        [cluster_id.to_s, cluster_id.to_s, {'data-max-auto-cores': max}]
+      end.sort_by { |option| option[0] }
+    end
+  end
+
+  def cluster_nodes
+    Rails.cache.fetch('script_cluster_nodes', expires_in: 4.hours) do 
+      {}.tap do |hash|
+        batch_clusters.map do |cluster|
+          hash[cluster.id] = cluster.job_adapter.nodes
+        end
+      end
+    end
+  end
+
+  def cluster_max_cores
+    Rails.cache.fetch('script_cluster_max_values', exipres_in: 4.hours) do
+      {}.tap do |hash|
+        cluster_nodes.each do |cluster_id, nodes|
+          hash[cluster_id] = nodes.max { |a, b| a.procs <=> b.procs }.procs
+        end
+      end
+    end
+  end
+
+  def batch_clusters
```

Comment:
Isn't this defined somewhere else? Seems like we need to refactor this to be shared?

Reply:
Is it? I can definitely see the value putting this in the configuration with `job_clusters` and `login_clusters`, but I can't find somewhere else that we're excluding the same types of clusters.

Reply:
Isn't it `SmartAttributes::AttributeFactory.batch_clusters` defined in `auto_batch_clusters.rb`? Seems like since it's a class method you can use that directly actually, so it's already shared even if it's in a weird spot.

Reply:
I had moved `batch_clusters` from `auto_batch_clusters.rb` to `cluster_cache.rb` because it's only really being used by the other functions in that module. Should it be kept in `auto_batch_clusters.rb` instead?

Reply:
> Should it be kept in auto_batch_clusters.rb instead?

Oh I see, yea moving it here from `auto_batch_clusters.rb` I think is the right move.  Sorry! Didn't catch that bit, I thought it was redefined, not moved.